src/session/session_api.c

```diff
@@ -1583,55 +1584,103 @@ __wt_session_range_truncate(
     trunc_info->stop = stop;
     trunc_info->orig_start_key = orig_start_key;
     trunc_info->orig_stop_key = orig_stop_key;
-    if (uri != NULL)
-        trunc_info->uri = uri;
-    else if (start != NULL)
-        trunc_info->uri = start->internal_uri;
-    else {
-        WT_ASSERT(session, stop != NULL);
-        trunc_info->uri = stop->internal_uri;
-    }
+    trunc_info->uri = actual_uri;
+
+    /*
+     * We would like to use bounded cursors if possible, so check whether they are supported. We
```

Comment:
We can shorten this a bit:
"Don't use bounded cursors for FLCS as it isn't supported, additionally skip using them for complex types such as column groups and indexes. We can't check support for those complex types at this abstraction level."

Reply:
Sounds good, I will use this comment instead.